examples/bri-3/src/bri/zeroKnowledgeProof/services/blockchain/ethereum/ethereum.service.ts

```diff
@@ -0,0 +1,48 @@
+import { ethers } from 'hardhat';
+import { readFile, writeFile } from 'fs/promises';
+import { Injectable } from '@nestjs/common';
+import { IBlockchainService } from '../blockchain.interface';
+import { Contract } from 'ethers';
+
+@Injectable()
+export class EthereumService implements IBlockchainService {
+  public async deployContract(contractName: string): Promise<void> {
+    const ccsmContract = await ethers.getContractFactory(contractName);
+    const deployedCcsmContract = await ccsmContract.deploy();
+    await this.storeDeployedContractAddress(deployedCcsmContract.address);
+  }
+
+  async connectToContract(contractName: string): Promise<Contract> {
+    const ccsmContractAddress = await this.getDeployedContractAddress();
+    return await ethers.getContractAt(contractName, ccsmContractAddress);
+  }
+
+  public async storeAnchorHash(
+    contractName: string,
+    anchorHash: string,
+  ): Promise<void> {
+    const ccsmContract = await this.connectToContract(contractName);
+    await ccsmContract.setAnchorHash(anchorHash);
+  }
+
+  private async getDeployedContractAddress(): Promise<string> {
+    return JSON.parse(
+      (
+        await readFile(
+          './zeroKnowledgeArtifacts/blockchain/ethereum/artifacts/ccsmContractAddress.json',
+        )
+      ).toString(),
+    ).contractAddress;
+  }
+
+  private async storeDeployedContractAddress(contractAddress: string) {
```

Comment:
we ll probably keep these in a db later, per BPI or per Workgroup. Mil5 stuff, i ll open an issue